stackdriver.go

```diff
@@ -183,18 +184,33 @@ type Options struct {
 	// the Resource you set uniquely identifies this Go process.
 	DefaultMonitoringLabels *Labels
 
-	// Context allows users to provide a custom context for API calls.
+	// Context allows you to provide a custom context for API calls.
 	//
 	// This context will be used several times: first, to create Stackdriver
 	// trace and metric clients, and then every time a new batch of traces or
 	// stats needs to be uploaded.
 	//
 	// If unset, context.Background() will be used.
 	Context context.Context
+
+	// GetMonitoredResource may be provided to supply the details of the
+	// monitored resource dynamically based on the tags associated with each
+	// data point. Most users will not need need to set this, but should instead
+	// set the MonitoredResource field.
+	//
+	// See the documentation on the MonitoredResource field for guidance on the
+	// interaction between monitored resources and labels.
+	//
+	// The MonitoredResource field is ignored if this field is set to a non-nil
+	// value.
+	GetMonitoredResource func([]tag.Tag) monitoredresource.Interface
```

Comment:
Should this take an ViewData and also return a new ViewData and MonitorResource. User usually want to remove that tag probably.

Reply:
Also some people want to configure the project id as well. This is useful when export metrics on behalf of others.

Reply:
I made it take and return a []tag.Tag since I think this will accomplish the same thing without the complexity of having to clone and mutate a full ViewData. Regarding the project ID, it is part of the MonitoredResource.